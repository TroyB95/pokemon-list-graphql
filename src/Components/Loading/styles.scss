.loading {
  width: 100%;
  height: 100vh;

  display: flex;
  align-items: center;
  justify-content: center;

  &__image {
    animation: spin 1.5s infinite linear;

    width: 64px;
    height: 64px;

    margin-left: 30px;
  }
}

@keyframes spin {
  100% {
    transform: rotate(360deg);
  }
}
