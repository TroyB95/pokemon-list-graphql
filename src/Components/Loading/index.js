import React from "react";

import Pikachu from "../../Assets/Images/pikachu.svg";

import "./styles.scss";

function Loading() {
  return (
    <div className="loading">
      Loading...
      <img className="loading__image" src={Pikachu} alt="the Pokemon Pikachu" />
    </div>
  );
}

export default Loading;
