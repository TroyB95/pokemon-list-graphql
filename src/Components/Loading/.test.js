import React from "react";
import { shallow } from "enzyme";

import Loading from "./index";

describe("Loading Component", () => {
  it("should render loading text and a pikachu image", () => {
    const component = shallow(<Loading />);

    expect(component).toMatchSnapshot();
  });
});
