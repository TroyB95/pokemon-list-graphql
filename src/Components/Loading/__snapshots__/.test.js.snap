// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should render loading text and a pikachu image 1`] = `
<div
  className="loading"
>
  Loading...
  <img
    alt="the Pokemon Pikachu"
    className="loading__image"
    src="pikachu.svg"
  />
</div>
`;
